### PR TITLE
Slcan review

### PR DESCRIPTION
**Pull request by albertoscotta**
General review of slcan.py
- add read method to enhance abstraction
- add flush method
- add get_serial_number, to retrive the serial number of the CANbus-serial bridge (also add test case)
- add get_version, to retrive the version of the CANbus-serial bridge (also add test case)
- wrap up stuff related to bitrate setting in dedicated methods
- make write method private/hidden

### CODE REVIEW

File: can/interfaces/slcan.py [hunk 1]
```diff
@@ -95,90 +96,113 @@ def __init__(
...
         frame = []
 
-        # First read what is already in the receive buffer
-        while (
-            self.serialPortOrig.in_waiting and self.LINE_TERMINATOR not in self._buffer
-        ):
-            self._buffer += self.serialPortOrig.read(1)
+        string = self.read(timeout)
 
-        # If we still don't have a complete message, do a blocking read
```
Comment (felixdivo) [inline can/interfaces/slcan.py]: Why is this not required (any more) and can be replaced with a simple read?

File: can/interfaces/slcan.py [hunk 2]
```diff
@@ -95,90 +96,113 @@ def __init__(
...
+    def set_bitrate_reg(self, btr):
+        self.close()
+        self.write("s" + btr)
+        self.open()
+
     def write(self, string):
         self.serialPortOrig.write(string.encode() + self.LINE_TERMINATOR)
         self.serialPortOrig.flush()
 
+    def read(self, timeout):
```
[the remark this answers is not in the document]
Comment (albertoscotta) [reply can/interfaces/slcan.py]: No. You're right! It's perfectly fine hidden

**Comment by felixdivo** [comment]
While we're at it: shouldn't `write()` be hidden as well? It might need deprecation thought.

cc @hardbayte

Apart from that we can merge this.

**Comment by albertoscotta** [comment]
> While we're at it: shouldn't `write()` be hidden as well? It might need deprecation thought.
> 
> cc @hardbyte
> 
> Apart from that we can merge this.

You know, I was thinking the same yesterday when I was trying to amend for you comments, but I didn't want to go too far.
If 'read' is hidden, I reckon cleaner to have its counterpart 'write' hidden as well.
I will push a commit for it. If it is not all right I will just revert it.

**Comment by hardbyte** [comment]
I think it is fine without a depreciation period - `write` should never have been exposed, and it isn't documented [here](https://python-can.readthedocs.io/en/master/interfaces/slcan.html)

**Comment by felixdivo** [comment]
Nice! Thanks again @albertoscotta 😄

**Pull request merged**